pkg/telemetry/vector.go

```diff
@@ -114,11 +115,14 @@ target = "stdout"
 			return "", err
 		} else {
 			if toml, found := vectorConfigMap.Data["vector.toml"]; found {
-				sinks = toml
+				vectorConfigToml = toml
 			} else {
 				return "", fmt.Errorf("vector.toml not found in config map %s", telemetrySpec.Vector.Config.Name)
 			}
 		}
+	} else if telemetrySpec.Vector.Components != nil {
```

Comment:
We discussed this briefly in private, but I think it would be less complex to just allow for the new way of configuring Vector via the CRD/Components. Unless we need to also support the original way of configuring it through a ConfigMap, I'd prefer the Component configuration you have added here.